ddev/src/ddev/cli/size/common.py

```
@@ -0,0 +1,153 @@
+    elif included_folder not in path:
+        return False
+    # It is an irrelevant file
+    elif any(ignore in path for ignore in ignored_files):
+        return False
+    # This file is contained in .gitignore
+    elif any(ignore in path for ignore in git_ignore):
+        return False
+    else:
+        return True
+
+
+def is_correct_dependency(platform, version, name):
+    return platform in name and version in name
+
+
+def print_csv(app, i, modules):
+    headers = modules[0].keys()
+    if i == 0:
+        app.display(",".join(headers))
+
+    for row in modules:
+        app.display(",".join(format(str(row[h])) for h in headers))
+
+
+def format(s):
+    if "," in s:
+        return '"' + s + '"'
+    else:
+        return s
+
+
+def print_table(app, modules, platform, version):
+    modules_table = {col: {} for col in modules[0].keys()}
+    for i, row in enumerate(modules):
+        for key, value in row.items():
+            modules_table[key][i] = str(value)
+    app.display_table(platform + " " + version, modules_table)
+
+
+def get_dependencies_sizes(app, deps, download_urls):
+    file_data = []
+    for dep, url in zip(deps, download_urls, strict=False):
+        dep_response = requests.head(url)
+        if dep_response.status_code != 200:
+            app.display_error(f"Error {dep_response.status_code}: Unable to fetch the dependencies file")
+            sys.exit(1)
+        else:
+            size = dep_response.headers.get("Content-Length", None)
+            file_data.append({"File Path": dep, "Type": "Dependency", "Name": dep, "Size (Bytes)": int(size)})
+
+    return file_data
+
+
+def get_dependencies(app, file_path):
+    download_urls = []
+    deps = []
+    try:
+        with open(file_path, "r", encoding="utf-8") as file:
+            file_content = file.read()
+            for line in file_content.splitlines():
+                match = re.search(r"([\w\-\d\.]+) @ (https?://[^\s#]+)", line)
+                if match:
```

Maybe we should throw an exception if the line doesn't match. We might miss some important data.